CMakeLists.txt

```diff
@@ -23,6 +23,7 @@ include(CheckIncludeFile)
 include(CheckFunctionExists)
 include(CheckSymbolExists)
 include(CheckStructHasMember)
+include(CTest)
```

Comment:
Could you describe, for CMake illiterates such as myself, in the commit message what running CTest entails? Do I have to install a separate program for that, or is CMake enough? (I vaguely seem to remember that this required Python to be installed?)

Speaking of the commit message: it is common practice to use a prefix in the first line. In this case, `cmake: ` strikes me as a sensible prefix. I would suggest to do that for every commit in this PR (except maybe the first one). This here commit message would thus begin with the line:

> cmake: add support for running the test suite using CTest

Reply:
There is nothing special about CTest. It ships with CMake itself. All it does in this case is 
cd t/ && sh t(something).sh
The advantage ctest has over prove is that it generates logs for the tests(stderr), which can be easy to diagnose.

Reply:
As I mentioned, please do describe this well in the commit message.

Reply:
Oh, and we could use this in the CI build, too, instead of using the home-rolled `test-helper run-command test-suite` thing.

Reply:
we could, but I don't understand the parallel strategy and other stuff related to the CI.
To do this we should also add the CTestTestfile.cmake (file that cmake generates for running the tests).
Side note: Bash on windows is about 3-4 times slower than linux. Something has to be done about this.

Reply:
>  Side note: Bash on windows is about 3-4 times slower than linux. Something has to be done about this.

Oh my, didn't I try. It is a _pain_! If you have ideas how to bring that about, I'm all ears.

Reply:
Give me some time, I am trying to compile bash without the msys/cygwin layers natively for windows with the help of gnulib. Will let you know about the results when it is ready.

Reply:
> we could, but I don't understand the parallel strategy and other stuff related to the CI.

Right, but that's easy. Basically, https://github.com/git-for-windows/git/blob/v2.26.1.windows.1/.github/workflows/main.yml#L152-L153 lists all the test scripts, ordered according to file size in descending order, and then uses round robin to distribute them onto the available agents. Currently, we distribute the load onto 10 agents in parallel. The biggest test script (as of time of writing, `t/t6043-merge-rename-directories.sh`) is run on the first agent, the next-biggest one by the second, and so on, the eleventh-biggest is run by the first agent again, etc. Essentially, the agents are assigned a number between 0 and 9, and all the tests are assigned numbers 0, ..., 900 (in descending file size order), and agent `nr` is running all tests where index `i` matches `i modulo 10 == nr`.